In [1]:
import pandas as pd
import numpy as np

In [2]:
mm= pd.read_table("../data/mm_scores_w_cornell.txt", dtype=str, sep='\t') #importing metadata file as pandas
mm.shape

(290, 25)

257 (orignal MM cohort) + 23 (cornell) = 290

In [3]:
mm.columns

Index([u'DateSurvey', u'IPAddress', u'DateSampled', u'MM_ID', u'KitID',
       u'SurveyID', u'LOTR_SCORE', u'SWB_SCORE', u'FLOURISH_SCORE',
       u'PANAST_POSAFFECT_SCORE', u'PANAST_NEGAFFECT_SCORE', u'BDI21_SCORE',
       u'LONE_SCORE', u'PSS_SCORE', u'STAITX_SCORE', u'NORM_LOTR_SCORE',
       u'NORM_SWB_SCORE', u'NORM_FLOURISH_SCORE',
       u'NORM_PANAST_POSAFFECT_SCORE', u'NORM_PANAST_NEGAFFECT_SCORE',
       u'NORM_BDI21_SCORE', u'NORM_LONE_SCORE', u'NORM_PSS_SCORE',
       u'NORM_STAITX_SCORE', u'BIN_BDI21_SCORE'],
      dtype='object')

In [4]:
ag_all= pd.read_table("../data/ag_all_plus_mm_missing.txt", sep='\t', index_col=0)
ag_all.head(5)

/Users/priya/anaconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.py:2723: DtypeWarning: Columns (167,168,169,170,171,172) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


acid_reflux acne_medication acne_medication_otc  \
10317.000027826     no_data           false                true   
10317.000027827     no_data           false               false   
10317.000027825     no_data           false               false   
10317.000014400     no_data            true                true   
10317.000027820     no_data           false               false   

                                     add_adhd age_cat age_corrected age_years  \
10317.000027826  I do not have this condition     30s          35.0      35.0   
10317.000027827  I do not have this condition     70+          77.0      77.0   
10317.000027825  I do not have this condition     30s          31.0      31.0   
10317.000014400                       Unknown     50s          50.0      50.0   
10317.000027820  I do not have this condition     30s          33.0      33.0   

                alcohol_consumption              alcohol_frequency  \
10317.000027826                true     Rarely (a few times/month)   
10317.000027827                true     Rarely (a few times/month)   
10317.000027825               false                          Never   
10317.000014400                true                          Daily   
10317.000027820                true  Occasionally (1-2 times/week)   

                alcohol_types_beercider          ...           \
10317.000027826                    true          ...            
10317.000027827                   false          ...            
10317.000027825                   false          ...            
10317.000014400                   false          ...            
10317.000027820                    true          ...            

                vioscreen_xylitol vioscreen_zinc  \
10317.000027826       0.037534246    9.568932604   
10317.000027827           Unknown        Unknown   
10317.000027825           Unknown        Unknown   
10317.000014400           Unknown        Unknown   
10317.000027820           Unknown        Unknown   

                vitamin_b_supplement_frequency vitamin_d_supplement_frequency  \
10317.000027826                          Daily                          Daily   
10317.000027827                          Daily                          Daily   
10317.000027825     Rarely (a few times/month)     Rarely (a few times/month)   
10317.000014400                        Unknown                        Unknown   
10317.000027820                          Daily                          Daily   

                vivid_dreams    weight_change weight_kg weight_units  \
10317.000027826      no_data  Remained stable      52.0    kilograms   
10317.000027827      no_data  Remained stable      68.0    kilograms   
10317.000027825      no_data  Remained stable      70.0    kilograms   
10317.000014400      no_data  Remained stable      59.0    kilograms   
10317.000027820      no_data  Remained stable      54.0    kilograms   

                                   whole_eggs whole_grain_frequency  
10317.000027826                         Never                 Never  
10317.000027827  Rarely (less than once/week)                 Never  
10317.000027825  Rarely (less than once/week)                 Daily  
10317.000014400                       Unknown               Unknown  
10317.000027820    Regularly (3-5 times/week)                 Daily  

[5 rows x 463 columns]

In [5]:
#change all column names to lower case
mm.columns = map(str.lower, mm.columns)

In [6]:
#removing ipaddress column
del mm['ipaddress']

In [7]:
#removing time from datesurvey
for x in range(len(mm['datesurvey'])):
    mm['datesurvey'][x]=mm['datesurvey'][x].split(' ')[0]

In [8]:
#Kit IDs are actually sample IDs
mm=mm.rename(columns={"kitid":"#SampleID"}) 
# changing sample ids to agp format
mm["#SampleID"]='10317.0000'+mm["#SampleID"]

In [9]:
mm=mm.set_index('#SampleID')

In [10]:
mm.head(5)

datesurvey datesampled mm_id surveyid lotr_score swb_score  \
#SampleID                                                                    
10317.000017083    5/24/14     5/27/14     1     4453         24       4.5   
10317.000017080    5/13/14     5/28/14     3     9801         18       3.5   
10317.000017096    5/24/14     5/28/14     4     6619         20       5.5   
10317.000017074    5/16/14     5/28/14     5     5675         19         5   
10317.000017103    5/28/14     5/28/14     6     4015          6       3.5   

                flourish_score panast_posaffect_score panast_negaffect_score  \
#SampleID                                                                      
10317.000017083             48                     26                     16   
10317.000017080             56                     46                     24   
10317.000017096             48                     36                     26   
10317.000017074             35                     26                     24   
10317.000017103             30                     24                     32   

                bdi21_score       ...       norm_lotr_score norm_swb_score  \
#SampleID                         ...                                        
10317.000017083           6       ...                   0.2           0.36   
10317.000017080           1       ...                   0.4            0.5   
10317.000017096           1       ...                  0.33           0.21   
10317.000017074           2       ...                  0.37           0.29   
10317.000017103          31       ...                   0.8            0.5   

                norm_flourish_score norm_panast_posaffect_score  \
#SampleID                                                         
10317.000017083                0.17                         0.6   
10317.000017080                   0                         0.1   
10317.000017096                0.17                        0.35   
10317.000017074                0.44                         0.6   
10317.000017103                0.54                        0.65   

                norm_panast_negaffect_score norm_bdi21_score norm_lone_score  \
#SampleID                                                                      
10317.000017083                        0.15              0.1            0.62   
10317.000017080                        0.35             0.02            0.08   
10317.000017096                         0.4             0.02            0.02   
10317.000017074                        0.35             0.03               0   
10317.000017103                        0.55             0.49            0.48   

                norm_pss_score norm_staitx_score bin_bdi21_score  
#SampleID                                                         
10317.000017083           0.36              0.36               1  
10317.000017080            0.2              0.53               1  
10317.000017096           0.14               0.6               1  
10317.000017074           0.32              0.89               1  
10317.000017103           0.82              0.76               4  

[5 rows x 23 columns]

In [11]:
notmm=[]
for idx in ag_all.index:
    if idx not in list(mm.index):
        notmm.append(idx)
ag_all=ag_all.drop(notmm)

In [12]:
ag_w_mm_scores=pd.concat([ag_all, mm], axis=1, join='inner')
ag_w_mm_scores.shape

(278, 486)

So now we have a metadata file with 278 samples (267 from MM + 11 from cornell)

In [13]:
mm_samples=list(mm.index)
overlapped_samples=list(ag_w_mm_scores.index)
c=0
print("The MM samples which are not included currently are:")
for samp in mm_samples:
    if samp not in overlapped_samples:
        print samp

The MM samples which are not included currently are:
10317.000026117
10317.000030767
10317.000017363
10317.000017365
10317.000017366
10317.000017374
10317.000017376
10317.000017377
10317.000017388
10317.000017389
10317.000017390
10317.000017407


In [15]:
ag_w_mm_scores.head(5)
ag_w_mm_scores=ag_w_mm_scores.reset_index()

In [16]:
ag_w_mm_scores.to_csv("../data/ag278_w_scores.txt", sep='\t')